File: Assets/scenes/warzonescene/script/BigExplosionBehaviour.js
#pragma strict

var maxSize : float = 100.0f;
var duration : float = 2.0f;
var sizeOverTime : AnimationCurve;
var alphaOverTime : AnimationCurve;

private var startTime : float;
private var exploding : boolean = false;
private var renderers : Renderer[];

var test : boolean = false;
function Start () {
	renderers = GetComponentsInChildren.<Renderer>();
}

function Explode(){
	startTime = Time.fixedTime;
	exploding = true;
}

function FixedUpdate () {
	if(test){
		test = false;
		Explode();
	}

	if(exploding){
		var t : float = (Time.fixedTime - startTime) / duration;
		transform.localScale = Vector3.one * sizeOverTime.Evaluate(t) * maxSize;
		for (var r : Renderer in renderers){
			r.material.color.a = alphaOverTime.Evaluate(t);
		}
		if(t > 1.0f){
			exploding = false;
			Destroy(gameObject);
		}
	}
	
}

File: Assets/scenes/warzonescene/script/BigExplosionBehaviour.cs
using UnityEngine;
using System;


public class BigExplosionBehaviour:MonoBehaviour{
	
	public float maxSize = 100.0f;
	public float duration = 2.0f;
	public AnimationCurve sizeOverTime;
	public AnimationCurve alphaOverTime;
	
	float startTime;
	bool exploding = false;
	Renderer[] renderers;
	
	public bool test = false;
	public void Start() {
		renderers = GetComponentsInChildren<Renderer>();
		foreach(Renderer r in renderers){
			r.enabled = false;
		}
	}
	
	public void Explode(){
		startTime = Time.fixedTime;
		exploding = true;
		foreach(Renderer r in renderers){
			r.enabled = true;
		}
	}
	
	public void FixedUpdate() {
		if(test){
			test = false;
			Explode();
		}
	
		if(exploding){
			float t = (Time.fixedTime - startTime) / duration;
			transform.localScale = Vector3.one * sizeOverTime.Evaluate(t) * maxSize;
			foreach(Renderer r in renderers){
				var tmp_cs1 = r.material.color;
                tmp_cs1.a = alphaOverTime.Evaluate(t);
                r.material.color = tmp_cs1;
			}
			if(t > 1.0f){
				exploding = false;
				Destroy(gameObject);
			}
		}
		
	}
}